test/test_testing.py

```diff
@@ -3080,6 +3080,45 @@ def test_stop_stopped_service(self):
         self.assertEqual(pebble.ServiceStartup.ENABLED, foo_info.startup)
         self.assertEqual(pebble.ServiceStatus.ACTIVE, foo_info.current)
 
+    def test_send_signal(self):
+        client = self.get_testing_client()
+        client.add_layer('foo', '''\
+        summary: foo
+        services:
+          foo:
+            summary: Foo
+            startup: enabled
+            command: '/bin/echo foo'
+          bar:
+            summary: Bar
+            command: '/bin/echo bar'
+        ''')
+        client.autostart_services()
+        # Foo is now started, but Bar is not
+
+        # Send a valid signal to a running service
+        client.send_signal("SIGINT", "foo")
```

Comment:
I had here SIGHUP before, and one of the tests on Windows failed because of it.

- At runtime, ops doesn't validate signal name - it is passed to pebble and pebble will bubble it back up if it's not valid. (And pebble obviously only runs on Linux.)
- For unittests, `send_signal` attempts to mock that response from pebble - I have it looking at `signal.Signal`, but it contains different things on Windows.

So if we're still supporting ops unittests on Windows, I should probably remove the signal name validation.

Reply:
Maybe we mark the pebble signal tests for skipping on windows.

Reply:
Maybe I'm missing something:
- `class Signals(IntEnum)` is going to have a different (lower) number of signals on Windows.
- For that reason, pebble.py will probably raise an exception on `sig = signal.Signals(sig).name` if you run on Windows and try to use a signal number that doesn't exist on Windows, for example SIGHUP.
- For that reason, we need to either modify pebble.py to convert against Linux sigs, not Windows sigs, or drop Windows support.

Reply:
I think that it would make sense to just skip this test for Windows.

This is in the Container class, which means that we assume Pebble to be present, and it doesn't make sense to break OF tests for an OS where Pebble doesn't (yet) run.

I don't think this counts as "dropping" windows support, though. We don't support Windows containers, but we can still do our best to support Windows elsewhere in the charm ecosystem.

Reply:
It's not just a windows problem. Here's a snip from `signal.Signal`:
```
    if sys.platform != "darwin" and sys.platform != "win32":
        SIGPOLL: int
        SIGPWR: int
```
iiuc, we need a `@unittest.skipUnless(is_linux, 'Pebble runs on Linux')` here. Correct?

Reply:
Seems reasonable for now.